sys/include/string_utils.h

```diff
@@ -38,6 +38,64 @@
 extern "C" {
 #endif
 
+/**
+ * @brief   String Writer structure.
+ *          Helper for writing multiple formatted strings to a buffer
+ */
+typedef struct {
+    const char *start;  /**< start of the target buffer */
+    char *ptr;          /**< current write pointer      */
```

Comment:
```suggestion
    char *position;          /**< current write pointer      */
```

(I think `ptr` is fine as it, but hey, maybe you find this better?)

Reply:
how about `head`